crypto/signature/ed25519.go

```diff
@@ -1,19 +1,112 @@
 package signature
 
 import (
+	"encoding/hex"
+	"fmt"
 	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ed25519"
 )
 
-// TODO: insert real implementations
-
-func SignEd25519(privateKey primitives.Ed25519PrivateKey, data []byte) primitives.Ed25519Sig {
-	return []byte{0x77}
+type Ed25519Signer struct {
+	publicKey        primitives.Ed25519PublicKey
+	privateKeyUnsafe primitives.Ed25519PrivateKey
 }
 
-func VerifyEd25519(publicKey primitives.Ed25519PublicKey, data []byte, signature primitives.Ed25519Sig) bool {
-	if signature.Equal([]byte{0x77}) {
-		return true
+const (
+	PUBLIC_KEY_SIZE  = 32
+	PRIVATE_KEY_SIZE = 64
+)
+
+// TODO NewFromRandUnsafe() this should probably be removed in the future, used for debugging mainly
+func NewFromRandUnsafe() (*Ed25519Signer, error) {
+	if pub, pri, err := ed25519.GenerateKey(nil); err != nil {
+		return nil, errors.Wrapf(err, "cannot create new signature from random keys")
 	} else {
-		return false
+		return &Ed25519Signer{
+			publicKey:        primitives.Ed25519PublicKey(pub),
+			privateKeyUnsafe: primitives.Ed25519PrivateKey(pri),
+		}, nil
+	}
+}
+
+func NewEd25519SignerPublicKeyString(pk string) (*Ed25519Signer, error) {
```

Comment:
Please let's not have args called `pk`, it's very confusing. You also use them some times for public and some times for private :)

Let's rename them to `publicKey` and `privateKey` where applicable